### PR TITLE
Document Object Ctower Rot

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Another opportunity to cleanup a couple of my first actors :).

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Ctower_Gear/z_bg_ctower_gear.c [hunk 1]
```diff
@@ -198,18 +178,19 @@ void BgCtowerGear_UpdateOrgan(Actor* thisx, GlobalContext* globalCtx) {
     }
 }
 
-// Using BgCtowerGear *this = THIS causes regalloc issues
 void BgCtowerGear_Draw(Actor* thisx, GlobalContext* globalCtx) {
-    func_800BDFC0(globalCtx, D_80AD32E8[BGCTOWERGEAR_GET_TYPE(thisx)]);
+    static Gfx* sDLists[] = { gClockTowerCeilingCogDL, gClockTowerCenterCogDL, gClockTowerWaterWheelDL };
+
+    func_800BDFC0(globalCtx, sDLists[BGCTOWERGEAR_GET_TYPE(thisx)]);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Ctower_Gear/z_bg_ctower_gear.c:184]: I'm not a huge fan of having in-function static data if it isn't required to match. I would personally prefer to put it outside of the function, but up to you.

File: src/overlays/actors/ovl_Bg_Ctower_Rot/z_bg_ctower_rot.h [hunk 2]
```diff
@@ -9,9 +9,9 @@ typedef void (*BgCtowerRotActionFunc)(struct BgCtowerRot*, GlobalContext*);
 
 typedef enum {
     /* 0 */ CORRIDOR,
-    /* 1 */ MAIN_DOOR, // Main door controls cutscene flow (updating Link fields and playing sounds). It is the one on
-                       // the left when entering from the Lost Woods
-    /* 2 */ DOOR,
+    /* 1 */ STONE_DOOR_MAIN, // Main door controls cutscene flow (updating Link fields and playing sounds). 
+                            // It is the one on the left when entering from the Lost Woods
+    /* 2 */ STONE_DOOR,
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Ctower_Rot/z_bg_ctower_rot.h:14]: What about prefixing those with something like `BGCTOWER_` or similar?

**Review by AngheloAlf — COMMENTED** [review]
Nice work with this object and the cleanup!

File: src/overlays/actors/ovl_Bg_Ctower_Rot/z_bg_ctower_rot.h (hunk 2 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_Bg_Ctower_Rot/z_bg_ctower_rot.h:14]: Especially since they're in the header.

Comment (hensldm) [reply src/overlays/actors/ovl_Bg_Ctower_Rot/z_bg_ctower_rot.h]: Ah, I did it for `Gear`, but must have forgot for `Rot`. Prefixed with `BGCTOWERROT_`.

**Review by AngheloAlf — APPROVED** [review]

**Comment by hensldm** [comment]
Just want to note that I've seen some comments about what object files should have, so have periodically added them to this PR. Namely using tlut instead of palette, adding a file header, using `empty#DL` instead of `emptyDL#`.

**Review by jpburnett — APPROVED** [review]
Nice documentation!

**Review by Kenix3 — APPROVED** [review]
Great formatting in your xml!

**Pull request merged**